server/protocols/webadmin/webadmin-mailqueue/src/test/java/org/apache/james/webadmin/routes/MailQueueRoutesTest.java

```diff
@@ -477,55 +512,115 @@ public void forcingDelayedMailsDeliveryShouldRejectRequestWithoutBody() throws E
     }
 
     @Test
     public void deleteMailsShouldDeleteMailsWhenNameIsGiven() throws Exception {
         MemoryMailQueue queue = mailQueueFactory.createQueue(FIRST_QUEUE);
-        String name = "mailName";
-        queue.enQueue(Mails.defaultMail()
-                .name(name)
+
+        queue.enQueue(FakeMail.builder()
+                .name(FAKE_MAIL_NAME_1)
                 .build());
 
+        queue.enQueue(FakeMail.builder()
+                .name(FAKE_MAIL_NAME_2)
+                .build());
+
+        String taskId = with()
+                .param("name", FAKE_MAIL_NAME_1)
+                .delete(FIRST_QUEUE + "/mails")
+                .jsonPath()
+                .getString("taskId");
+
         given()
-            .param("name", name)
+            .basePath(TasksRoutes.BASE)
         .when()
-            .delete(FIRST_QUEUE + "/mails")
+            .get(taskId + "/await")
         .then()
-            .statusCode(HttpStatus.NO_CONTENT_204);
-
-        assertThat(queue.browse()).isEmpty();
+            .body("status", is("completed"))
+            .body("taskId", is(notNullValue()))
+            .body("type", is(DeleteMailsFromMailQueueTask.TYPE))
+            .body("additionalInformation.mailQueueName", is(FIRST_QUEUE))
+            .body("additionalInformation.initialCount", is(2))
+            .body("additionalInformation.remainingCount", is(1))
+            .body("additionalInformation.name", is(FAKE_MAIL_NAME_1))
+            .body("startedDate", is(notNullValue()))
+            .body("submitDate", is(notNullValue()))
+            .body("completedDate", is(notNullValue()));
     }
 
     @Test
     public void deleteMailsShouldDeleteMailsWhenRecipientIsGiven() throws Exception {
         MemoryMailQueue queue = mailQueueFactory.createQueue(FIRST_QUEUE);
-        String recipient = "recipient@james.org";
-        queue.enQueue(Mails.defaultMail()
-                .recipient(recipient)
+
+        queue.enQueue(FakeMail.builder()
+                .name(FAKE_MAIL_NAME_1)
+                .recipient(RECIPIENT_JAMES_ORG)
+                .build());
+
+        queue.enQueue(FakeMail.builder()
+                .name(FAKE_MAIL_NAME_2)
+                .recipient(RECIPIENT_JAMES_ORG)
+                .build());
+
+        queue.enQueue(FakeMail.builder()
+                .name(FAKE_MAIL_NAME_2)
+                .recipient(RECIPIENT_1_JAMES_ORG)
                 .build());
 
+        String taskId = with()
+                .param("recipient", RECIPIENT_JAMES_ORG)
+                .delete(FIRST_QUEUE + "/mails")
+                .jsonPath()
+                .getString("taskId");
+
         given()
-            .param("recipient", recipient)
+            .basePath(TasksRoutes.BASE)
         .when()
-            .delete(FIRST_QUEUE + "/mails")
+            .get(taskId + "/await")
         .then()
-            .statusCode(HttpStatus.NO_CONTENT_204);
-
-        assertThat(queue.browse()).isEmpty();
```

Comment:
And here also...

Reply:
Note that the test about `additionalInformation` needs to be done separately (in a new test)